### PR TITLE
BLOCKCHAIN-463 - Judiciary Phase 1

### PR DESCRIPTION
**Pull request by kacperzuk-neti**
Process from merging to forcing move of some staked llm on real chain:
1. Release new runtime version, upgrade the chain
2. Deploy the contract with the initial set of judges
3. sudo call llm.setCourts, providing address of the contract as a court
4. First judge calls `propose` with `Proposal::LLMForceTranfer {...}` on the contract
5. Remaining judges (as many as required by the configured threshold) call `approve`
6. Proposal will execute automatically when threshold is reached.

### CODE REVIEW

File: substrate/frame/llm/src/lib.rs [hunk 1]
```diff
@@ -507,6 +385,61 @@ pub mod pallet {
...
+					let politics_balance = LLMPolitics::<T>::get(account.clone());
+					ensure!(politics_balance >= amount, Error::<T>::LowBalance);
+
+					LLMPolitics::<T>::mutate(account.clone(), |b| *b -= amount);
+					Self::deposit_event(Event::<T>::LLMPoliticsUnlocked(account, amount));
+					politipool_account.clone()
+				},
+			};
+			let transfer_to = match to {
+				LLMAccount::Liquid(account) => account,
```
Comment (kacperzuk-neti) [inline substrate/frame/llm/src/lib.rs:422]: This means that Courts are able to instantly unlock staked LLM. Is that OK or do we want only staked -> staked moves?

File: contracts/msig_court/lib.rs [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+	use ink::prelude::vec::Vec;
+	use ink::storage::Mapping;
+	use liberland_extension::LLMForceTransferArguments;
+
+	#[derive(Debug, Clone, PartialEq, Eq)]
+	#[ink::scale_derive(Encode, Decode, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(ink::storage::traits::StorageLayout))]
+	pub enum Proposal {
+		LLMForceTransfer(LLMForceTransferArguments),
+		SetGovernance { threshold: u32, judges: Vec<AccountId> },
```
Comment (kacperzuk-neti) [inline contracts/msig_court/lib.rs:18]: There's no such thing as sudo/root for contracts. So instead, adding/removing judges can be made by current judges, just like any other proposal.

If that's not OK and we want sudo, we can add a contract owner, set it to some address with no private key then use sudo.sudoAs to trigger it. Let me know what you prefer and I'll adjust if needed.

File: substrate/frame/llm/src/lib.rs (hunk 1 above)
Comment (DorianSternVukotic) [reply substrate/frame/llm/src/lib.rs:424]: staked -> staked for now

File: contracts/msig_court/lib.rs (hunk 2 above)
Comment (DorianSternVukotic) [reply contracts/msig_court/lib.rs:18]: Can senate or congress be contract owner (for now)? I expect making that tx would be cringe but doable...

Comment (kacperzuk-neti) [reply contracts/msig_court/lib.rs:18]: Right now they can't. We can use custom-account pallet for this, but we need to allow contract interactions in the call filter. It's trivial to do.
Should I add this? Do we want senate or congress? If congress, can senate veto?

File: substrate/frame/llm/src/lib.rs (hunk 1 above)
Comment (kacperzuk-neti) [reply substrate/frame/llm/src/lib.rs]: Changed :+1:

File: substrate/frame/llm/src/lib.rs [hunk 3]
```diff
@@ -163,8 +26,17 @@ type Assets<T> = pallet_assets::Pallet<T>;
...
+use sp_runtime::RuntimeDebug;
+
+#[derive(PartialEq, Eq, Clone, RuntimeDebug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+pub enum LLMAccount<AccountId> {
+	Liquid(AccountId),
+	Locked(AccountId),
+}
 
-#[frame_support::pallet]
+#[frame_support::pallet(dev_mode)] // FIXME
```
Comment (DorianSternVukotic) [inline substrate/frame/llm/src/lib.rs:39]: fixit

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]

File: substrate/frame/llm/src/lib.rs (hunk 3 above)
Comment (kacperzuk-neti) [reply substrate/frame/llm/src/lib.rs]: Done

**Pull request merged**